ui/src/components/Tasks/DataCollection.js

```diff
@@ -375,6 +375,12 @@ export default connect((state) => {
       ].acq_parameters.osc_range;
   }
 
+  // Use task data if samlpleID is a array (tasks added from the sample
+  // list view) or if we are using the option "rememberParametersBetweenSamples"
+  const useTaskData =
```

Comment:
I would maybe call this `withTaskData` to avoid confusion with React hooks.

Reply:
Also, I feel like this condition could be computed directly inside `toFixed` to avoid repetition. If the problem is that the hardware object name doesn't always match the form field name, then this can be solved by adding a third, optional argument:

```js
function toFixed(state, hoName, fieldName) {
  const withTaskData =
    state.taskForm.sampleIds.constructor !== Array ||
    state.queue.rememberParametersBetweenSamples;

  const value = withTaskData ? state.taskForm.taskData.parameters[fieldName] : state.beamline.hardwareObjects[hoName];

 // maybe a check that throws a clear error if `value` is `undefined` here

  let precision = ...;
  return value.toFixed(precision);
}
```

Reply:
> I would maybe call this `withTaskData` to avoid confusion with React hooks.

I knew that you would say that :), but yeah you are right, Naming by convention in frameworks not sure I'm a fan :) Anyways, withTaskData works as well :)

Reply:
Excellent idea @axelboc, so I finally did something in the lines of what you suggested, much nicer !